connector/mqtt/vernemq/v2k-brigde/app/MqttClient.js

```diff
@@ -1,68 +1,130 @@
 const fs = require('fs');
-const path = require('path');
+const async = require('async');
 const mqtt = require('mqtt');
-const uuid4 = require('uuid/v4');
 const { logger } = require('@dojot/dojot-module-logger');
 const defaultConfig = require('./config');
 
 const TAG = { filename: 'MqttClient' };
-const AgentMessenger = require('./AgentMessenger');
+
+/**
+ * BackPressure
+ */
 
 class MQTTClient {
-  constructor(config) {
+  constructor(agentMessenger, config) {
     this.config = config || defaultConfig;
     this.isConnected = false;
-    this.agentMessenger = null;
 
-    this.clientId = `${this.config.mqtt.clientId}-${uuid4()}`;
+    this.clientId = this.config.mqtt.clientId;
+    this.host = this.config.mqtt.host;
+    this.hostname = this.config.app.hostname;
+    this.keepAlive = this.config.mqtt.keepAlive;
+    this.port = this.config.mqtt.port;
     this.username = this.config.mqtt.clientUsername;
+    this.secureMode = this.config.mqtt.secure;
+
+    this.privateKey = fs.readFileSync(`${this.config.mqtt.tls.privateKey.location}`);
+    this.clientCrt = fs.readFileSync(`${this.config.mqtt.tls.certificate.location}`);
+    this.ca = fs.readFileSync(`${this.config.mqtt.tls.ca.location}`);
+
+    // backPressure
```

Comment:
Kafka producer implements a back-pressure mechanism (queue.buffering.backpressure.threshold). It is necessary to figure out how it works and whether we can integrate it with our implementation. It seems that our queue is redundant.

Reply:
Kafka producer doesn't externalize its backpressure mechanism to interact with.